model/src/main/kotlin/config/DownloaderConfiguration.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.config
+
+import org.ossreviewtoolkit.model.SourceCodeOrigin
+
+data class DownloaderConfiguration(
+    /**
+     * Configuration of the considered source code origins and their priority order.
+     */
+    val sourcesOrigin: List<SourceCodeOrigin> = listOf(SourceCodeOrigin.VCS, SourceCodeOrigin.ARTIFACT)
+) {
+    init {
+        if (sourcesOrigin.isEmpty()) {
+            throw InvalidSourcesOriginConfiguration("Empty 'sourcesOrigin' found, please specify ")
+        }
```

Comment:
In other similar places we simply use `require() {}`, so I believe we should do so here, too.